### PR TITLE
unicorn: support stop points that aren't starts of basic blocks

### PR DESCRIPTION
**Pull request by bennofs**
Previously, if a stop point was not placed on the start of a basic point, then
unicorn simply ignored it. We now check if the current basic block contains any
of the stop points, and if it does, stop execution.

This also includes a small fix to the `perf_unicorn.py` script to adapt to current version of angr (plus a check to launch a debugger if execution errored).

This is the first step towards supporting `explore` and breakpoints with unicorn.

### CODE REVIEW

File: tests/perf_unicorn.py [hunk 1]
```diff
@@ -33,6 +36,9 @@ def perf_unicorn_1():
     pg_unicorn.run()
     elapsed = time.time() - start
 
+    if len(pg_unicorn.errored) > 0:
+        pg_unicorn.errored[0].debug()
+
```
Comment (rhelmot) [inline tests/perf_unicorn.py]: Can you remove this and the above check? We'd like our tests to not hang on error :)

Comment (bennofs) [inline tests/perf_unicorn.py]: Oh, i did not think about that. Removed.

**Comment by rhelmot** [comment]
Alright, now that tests are almost reasonable again it looks like we can say that you've broken the oppologist. Can you look into fixing that?

**Comment by bennofs** [comment]
The new commit should fix that. Let's see what travis says

**Comment by bennofs** [comment]
Angr passes now.

**Comment by zardus** [comment]
I'm doing a retry for colorguard to check if that was an intermittent failure.

**Comment by zardus** [comment]
Yep -- looks to be in line with master!

**Pull request merged**